wfe/web-front-end.go

```diff
@@ -998,11 +1008,19 @@ func (wfe *WebFrontEndImpl) Certificate(response http.ResponseWriter, request *h
 			if strings.HasPrefix(err.Error(), "gorp: multiple rows returned") {
 				wfe.sendError(response, "Multiple certificates with same short serial", err, http.StatusConflict)
 			} else {
-				wfe.sendError(response, "Not found", err, http.StatusNotFound)
+				response.Header().Add("Cache-Control", "public, max-age=0, no-cache")
+				wfe.sendError(response, "Certificate not found", err, http.StatusNotFound)
 			}
 			return
 		}
 
+		// Set cache-control header if certificate NotAfter is > time.Now().Add(-WFE.CertCacheExpirationWindow)
+		if time.Now().Add(-wfe.CertNoCacheExpirationWindow).After(cert.Expires) {
```

Comment:
It's a safeguard against the cert changing at the end, and is for proxies less well behaved than our CDN, such as those which override max-age.
To be honest, I think we're fine however you want to do it. If we want to be precise, we can wait and see if anyone ever gets in such a state and look to change behavior then. That's fine.